app/src/main/cpp/skyline/services/nvdrv/devices/nvhost_as_gpu.cpp

```diff
@@ -43,107 +36,120 @@ namespace skyline::service::nvdrv::device {
             state.logger->Warn("Failed to unmap chunk at 0x{:X}", offset);
+
+        return NvStatus::Success;
     }
 
-    void NvHostAsGpu::Modify(IoctlData &buffer) {
+    NvStatus NvHostAsGpu::Modify(IoctlType type, std::span<u8> buffer, std::span<u8> inlineBuffer) {
         struct Data {
-            u32 flags;
-            u32 kind;
-            u32 nvmapHandle;
-            u32 pageSize;
-            u64 bufferOffset;
-            u64 mappingSize;
-            u64 offset;
-        } region = state.process->GetObject<Data>(buffer.input.at(0).address);
-
-        if (!region.nvmapHandle)
-            return;
-
-        auto nvmap = state.os->serviceManager.GetService<nvdrv::INvDrvServices>("nvdrv")->GetDevice<nvdrv::device::NvMap>(nvdrv::device::NvDeviceType::nvmap)->handleTable.at(region.nvmapHandle);
-
-        u64 mapPhysicalAddress = region.bufferOffset + nvmap->address;
-        u64 mapSize = region.mappingSize ? region.mappingSize : nvmap->size;
-
-        if (region.flags & 1)
-            region.offset = state.gpu->memoryManager.MapFixed(region.offset, mapPhysicalAddress, mapSize);
-        else
-            region.offset = state.gpu->memoryManager.MapAllocate(mapPhysicalAddress, mapSize);
+            u32 flags;        // In
+            u32 kind;         // In
+            u32 nvmapHandle;  // In
+            u32 pageSize;     // InOut
+            u64 bufferOffset; // In
+            u64 mappingSize;  // In
+            u64 offset;       // InOut
+        }  &data = util::As<Data>(buffer);
+
+        try {
+            auto driver = nvdrv::driver.lock();
+            auto nvmap = driver->nvMap.lock();
+            auto mapping = nvmap->handleTable.at(data.nvmapHandle);
+
+            u64 mapPhysicalAddress = data.bufferOffset + mapping->address;
+            u64 mapSize = data.mappingSize ? data.mappingSize : mapping->size;
+
+            if (data.flags & 1)
+                data.offset = state.gpu->memoryManager.MapFixed(data.offset, mapPhysicalAddress, mapSize);
+            else
+                data.offset = state.gpu->memoryManager.MapAllocate(mapPhysicalAddress, mapSize);
+
+            if (data.offset == 0) {
+                state.logger->Warn("Failed to map GPU address space region!");
+                return NvStatus::BadParameter;
+            }
 
-        if (region.offset == 0) {
-            state.logger->Warn("Failed to map GPU address space region!");
-            buffer.status = NvStatus::BadParameter;
+            return NvStatus::Success;
+        } catch (const std::out_of_range &) {
+            state.logger->Warn("Invalid NvMap handle: 0x{:X}", data.nvmapHandle);
+            return NvStatus::BadParameter;
         }
-
-        state.process->WriteMemory(region, buffer.output.at(0).address);
     }
 
-    void NvHostAsGpu::GetVaRegions(IoctlData &buffer) {
+    NvStatus NvHostAsGpu::GetVaRegions(IoctlType type, std::span<u8> buffer, std::span<u8> inlineBuffer) {
+        /*
```

Comment:
Whys this commented out?

Reply:
Unused at the moment, might be used later on.